library/network/src/modules/SuSEFirewall.rb

```diff
@@ -323,6 +323,71 @@ def GetServices(services)
       deep_copy(services_status)
     end
 
+    # Function returns map of supported services all network interfaces.
+    #
+    # @param	list <string> of services
+    # @return	[Hash <String, Hash{String => Boolean} >]
+    #
+    #
+    # **Structure:**
+    #
+    #    	Returns $[service : $[ interface : supported_status ]]
+    #
+    # @example
+    #	GetServicesInZones (["service:irc-server"]) -> $["service:irc-server":$["eth1":true]]
+    #  // No such service "something"
+    #	GetServicesInZones (["something"])) -> $["something":$["eth1":nil]]
+    #  GetServicesInZones (["samba-server"]) -> $["samba-server":$["eth1":false]]
+    def GetServicesInZones(services)
+      services = deep_copy(services)
+      # list of interfaces for each zone
+      interface_in_zone = {}
+
+      Builtins.foreach(GetListOfKnownInterfaces()) do |interface|
+        # zone of interface
+        zone_used = GetZoneOfInterface(interface)
+        # interface can be unassigned
+        next if zone_used.nil? || zone_used == ""
+        Ops.set(
+          interface_in_zone,
+          zone_used,
+          Builtins.add(Ops.get(interface_in_zone, zone_used, []), interface)
+        )
+      end
+
+      # $[ service : $[ network_interface : status ]]
+      services_status = {}
+
+      # Do the mapping here since we may have more than one FirewallD service
+      # per SF2 service
+      if self.class == SuSEFirewalld
```

Comment:
Doesn't comparing class names look suspicious?
In fact, we can move this code to the subclass:
```rb
class SuSEFirewalld
  def GetServicesInZones(services)
    services = services.map{ ... }.flatten
    super # equivalent to Firewall::GetServicesInZones(services)
  end
end